src/openrct2/world/Park.cpp

```diff
@@ -157,6 +159,20 @@ namespace OpenRCT2::Park
         }
 
         suggestedMaxGuests = std::min<uint32_t>(suggestedMaxGuests, 65535);
+
+#ifdef ENABLE_SCRIPTING
+        auto& hookEngine = GetContext()->GetScriptEngine().GetHookEngine();
+        if (hookEngine.HasSubscriptions(HOOK_TYPE::PARK_CALCULATE_GUEST_CAP))
+        {
+            auto ctx = GetContext()->GetScriptEngine().GetContext();
+            auto obj = DukObject(ctx);
+            obj.Set("suggestedGuestMaximum", suggestedMaxGuests);
+            auto e = obj.Take();
+            hookEngine.Call(HOOK_TYPE::PARK_CALCULATE_GUEST_CAP, e, true);
+
+            suggestedMaxGuests = AsOrDefault(e["suggestedGuestMaximum"], static_cast<int32_t>(suggestedMaxGuests));
```

Comment:
I think it would be a good idea to clamp the value to [0, 65535] at this point.

Reply:
I'd prefer to use std::limits with the type over hardcoded values, fwiw.

Reply:
That is a good idea

Reply:
Would that be using the type uint32_t?

Reply:
Is this correct?

```
suggestedMaxGuests = std::min(
            std::max(
                static_cast<uint32_t>(AsOrDefault(e["suggestedGuestMaximum"], static_cast<int32_t>(suggestedMaxGuests))),
                std::numeric_limits<uint32_t>::min()
            ),
            std::numeric_limits<uint32_t>::max()
);
```

Reply:
Look into std::clamp, that'll simplify things a lot :)

Reply:
Sorry, just saw your message now - will change it.